onnxruntime/core/session/inference_session.cc

```diff
@@ -1220,8 +1224,46 @@ Status AssignNodesToEpsFromHashes(Graph& graph, const fbs::SessionState& fbs_ses
   ORT_RETURN_IF_ERROR(VerifyEachNodeIsAssignedToAnEp(graph, logger));
   return Status::OK();
 }
-#endif  // defined(ENABLE_ORT_FORMAT_LOAD)
+
+Status TransformGraphForOrtFormatModel(onnxruntime::Graph& graph, const ExecutionProviders& providers,
+                                       KernelRegistryManager& kernel_registry_manager,
+                                       SessionState& session_state,
+                                       const experimental::fbs::SessionState& serialized_session_state,
+                                       const logging::Logger& logger) {
+#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
+  // nodes are already partitioned, but a custom EP may compile some at runtime.
+  // run the partitioning to allow that to happen.
+  //
+  // We always have the CPU EP, so only need to run this if some other EP is enabled
+  if (providers.NumProviders() > 1) {
+    ORT_RETURN_IF_ERROR(PartitionOrtFormatModel(graph, providers, kernel_registry_manager,
+                                                session_state));
+  }
+
+#if defined(ORT_ENABLE_ORT_FORMAT_RUNTIME_GRAPH_OPTIMIZATION)
```

Comment:
>ORT_ENABLE_ORT_FORMAT_RUNTIME_GRAPH_OPTIMIZATION [](http://example.com/codeflow?start=12&length=48)

Isn't this and !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD) the same thing?

Reply:
Should be a separate option as it will cost binary size and depending on your usage you may not need/want runtime optimization.